ravenframework/SupervisedLearning/SyntheticHistory.py

```diff
@@ -215,6 +215,68 @@ def findAlgoByName(self, name):
         return algo
     return None
 
+  def getFundamentalFeatures(self, requestedFeatures, featureTemplate=None):
+    """
+      Collect the fundamental parameters for this ROM
+      Used for writing XML, interpolating, clustering, etc
+      NOTE: This is originally copied over from SupervisedLearning!
+         Primarily using this for interpolation.
+      @ In, requestedFeatures, dict(list), featureSet and features to collect (may be None)
+      @ In, featureTemplate, str, templated string for naming features (probably leave as None)
+      @ Out, features, dict,
+    """
+    # NOTE: this should match the clustered features template.
+    if featureTemplate is None:
+      featureTemplate = '{target}|{metric}|{id}' # TODO this kind of has to be the format currently
+
+    requests = self._getClusterableFeatures()
+    features = self.getLocalRomClusterFeatures(featureTemplate, {}, requests, picker=None)
+
+    return features
+
+  def readFundamentalFeatures(self, features):
+    """
+      Reads in the requested ARMA model properties from a feature dictionary
+      @ In, features, dict, dictionary of fundamental features
+      @ Out, readFundamentalFeatures, dict, more clear list of features for construction
+    """
+    return features
+
+  def setFundamentalFeatures(self, features):
+    """
+      opposite of getFundamentalFeatures, expects results as from readFundamentalFeatures
+      Constructs this ROM by setting fundamental features from "features"
+      @ In, features, dict, dictionary of info as from readFundamentalFeatures
+      @ Out, None
+    """
+    # NOTE: we deepcopy'd a ROM to get here... so any non-clusterable features have
+    #       been copied over. For example: ARMA 'results', 'model', 'initials'
+    # TODO: these attributes should be overloaded in some fashion in the future
+    self.setLocalRomClusterFeatures(features)
+    self.amITrained = True
+
+  def parametrizeGlobalRomFeatures(self, featureDict):
+    """
+      Parametrizes the GLOBAL features of the ROM (assumes this is the templateROM and segmentation is active)
+      @ In, featureDict, dictionary of features to parametrize
```

Comment:
missing type in docstring after name (I'm sure it's missing in the original, too, sorry)